src/rcv/swiftnav.c

```diff
@@ -650,9 +837,107 @@ static int decode_gpsnav(raw_t *raw) {
   return 2;
 }
 
+/* decode SBP nav message for BDS (navigation data) --------------------------*/
+static int decode_bdsnav(raw_t *raw) {
+
+  uint8_t *puiTmp = (raw->buff) + 6;
+  eph_t eph = {0};
+  uint8_t prn, sat;
+
+  trace(4, "SBP decode_bdsnav: len=%d\n", raw->len);
+
+  if ((raw->len) < 147) {
+    trace(2, "SBP decode_bdsnav frame length error: len=%d\n", raw->len);
+    return -1;
+  }
+
+  prn = puiTmp[0];
+  if (!((prn >= 1) && (prn <= 37))) {
+    trace(2, "SBP decode_bdsnav prn error: sat=%d\n", prn);
+    return -1;
+  }
+
+  sat = satno(SYS_CMP, prn);
+  if (sat == 0)
+    return -1;
+
+  eph.code = puiTmp[1];
+  if ((CODE_BDS2_B11 != eph.code)) {
+    trace(2, "Unrecognised code %d for C%02d\n", eph.code, prn);
+    return -1;
+  }
+
+  decode_bdsnav_common(puiTmp - 2, &eph);
+
+  eph.ttr = raw->time;
+
+  if (!strstr(raw->opt, "EPHALL")) {
+    if ((eph.iode == raw->nav.eph[sat - 1].iode) &&
```

Comment:
Any suggestions for this part?
We don't have proper usage of iode & iodc from FW.

Reply:
For BDS that is.

Reply:
I don't believe that BDS has the concept of IODE, the age of corrections has been used but is not equivalent.